compiler/src/iree/compiler/Dialect/Encoding/IR/EncodingAttrs.td

```diff
@@ -63,8 +63,19 @@ def EncodingAttr :
     AttrParameter<"ArrayAttr", "element types of the user's operands">:$element_types,
     OptionalParameter<"ArrayAttr", "Indexing maps of the operation using this tensor">:$user_indexing_maps,
     OptionalParameter<"AffineMapAttr", "Indexing map that represents the broadcasting dims in the producer">:$bcast_map,
-    // TODO(hanchung): The round_dims_to parameter can be revisited. We explicitly map them to M,N,K dimension for now.
-    OptionalParameter<"DenseArrayAttr", "Values for padding M,N,K dimensions">:$round_dims_to
+    // TODO(hanchung): Deprecate the round_dims_to field when we plumb the layouts
+    // field through the whole stack. See https://github.com/iree-org/iree/issues/17924
+    // for details. Note that today we abuse the attribute to carry narrow
+    // matrix information. The end goal is deprecating the field and add a
+    // "iteration_space_size" field to describe the shape. It is useful to
+    // handle narrow matrix cases.
+    OptionalParameter<"DenseArrayAttr", "Values for padding M,N,K dimensions">:$round_dims_to,
+    OptionalParameter<"ArrayAttr", "An array of attributes that describes the "
+    "potential layouts on the device. It is an array because a device could "
+    "have several executable targets. Note that it can be any attribute that "
+    "implements EncodingLayoutAttrInterface. The expectation of the field "
```

Comment:
Now that we have an ArrayAttr that expects a specific type for its elements, it would probably be good to add a verifier to the EncodingAttr. I think we should be enforcing that the elements of `layouts` implement `EncodingLayoutAttrInterface`. The verifier doesn't have to be part of this PR, though, it can be added as a follow up.

Reply:
I might abuse the `layouts` field in some ways. I like the idea that we could discard them anytime. Think that we have a device with multiple executable targets. We are not able to resolve an EncodingLayoutAttrInterface attribute if the target does not implement it. We could attach a default attribute that behaves like dropping all the encodings, though. I'll think more about it when pictures are clearer to me. It might just a question about how we compose pieces together.